### DISCUSSION TITLE
[Security] What's the difference between IS_AUTHENTICATED and IS_AUTHENTICATED_REMEMBERED?

### DISCUSSION THREAD

**Discussion by artyuum** [symfony/symfony#47371] [Q&A] [answered]
First of all, I noticed that the `IS_AUTHENTICATED` attribute is not documented in the security pages and stumbled upon it thanks to the [AuthenticatedVoter](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Security/Core/Authorization/Voter/AuthenticatedVoter.php). I'm willing to open a PR about it if this discussion helps me to understand a bit more about it.

I saw that `IS_AUTHENTICATED` *seems* to act the same as `IS_AUTHENTICATED_REMEMBERED` because, from my understanding, all authenticated users will have both attributes but when I look into the code, the condition that is used on both attributes is different.

These attributes are used as follows in the AuthenticatedVoter and both call different methods from the [AuthenticationTrustResolver](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Security/Core/Authentication/AuthenticationTrustResolver.php):
https://github.com/symfony/symfony/blob/85065e4f76eb0e75163277498bbc4fadb3ae49f8/src/Symfony/Component/Security/Core/Authorization/Voter/AuthenticatedVoter.php#L69-L77

From the AuthenticationTrustResolver, we can see the following:
https://github.com/symfony/symfony/blob/b6382807912e0d67c092d7aae062d972003096ff/src/Symfony/Component/Security/Core/Authentication/AuthenticationTrustResolver.php#L24-L43

Is it safe to use the `IS_AUTHENTICATED` attribute to simply check if a user is logged in (no matter if they are logged in thanks to the "remember-me" cookie or not)?

**Comment by stof** [comment, marked as answer]
Yes, this is exactly what `IS_AUTHENTICATED` is about.

**Reply by artyuum** [reply]
In that case, what's the difference between both attributes? In what context should we use the attribute IS_AUTHENTICATED_REMEMBERED instead of IS_AUTHENTICATED and vice versa?

**Reply by stof** [reply]
`IS_AUTHENTICATED_REMEMBERED` is available since Symfony 2.0. `IS_AUTHENTICATED` was added in 5.3 (if I remember correctly) to make things easier to understand, as it better describes the intent when you just want to check that the user is authenticated.

**Reply by artyuum** [reply]
Ah, good to know. So that means there is a plan to deprecate the  IS_AUTHENTICATED_REMEMBERED attribute in favor of IS_AUTHENTICATED at some point?

**Reply by stof** [reply]
no such plan has been discussed for now

**Reply by artyuum** [reply]
I would like to propose this change (deprecate `IS_AUTHENTICATED_REMEMBERED` in favor `IS_AUTHENTICATED` if they are really identical) and know if it's a good idea or not. Should I create an issue or another discussion?

**Reply by MatTheCat** [reply]
Even if these are functionally similar (for now) their semantics are different. I don't think `IS_AUTHENTICATED_REMEMBERED` needs to be deprecated.
